java/client/src/org/openqa/selenium/devtools/network/Network.java

```diff
@@ -0,0 +1,521 @@
+    url.ifPresent(string -> params.put("domain", domain));
+    url.ifPresent(string -> params.put("path", path));
+
+    return new Command<>(domainName + ".deleteCookies", params);
+
+  }
+
+  /**
+   * Disables network tracking, prevents network events from being sent to the client.
+   * @return DevTools Command
+   */
+  public static Command<Void> disable() {
+    return new Command<>(domainName + ".disable", ImmutableMap.of());
+  }
+
+  /**
+   * Activates emulation of network conditions.
+   * @param offline - True to emulate internet disconnection.
+   * @param latency - Minimum latency from request sent to response headers received (ms).
+   * @param downloadThroughput - Maximal aggregated download throughput (bytes/sec). -1 disables download throttling.
+   * @param uploadThroughput - Maximal aggregated upload throughput (bytes/sec). -1 disables upload throttling.
+   * @param connectionType (Optional) - The underlying connection technology that the browser is supposedly using.
+   * @return DevTools Command
+   */
+  public static Command<Void> emulateNetworkConditions(boolean offline, double latency, double downloadThroughput, double uploadThroughput, Optional<ConnectionType> connectionType) {
+
+    Map<String, Object> params = new HashMap<>();
+
+    params.put("offline", offline);
+    params.put("latency", latency);
+    params.put("downloadThroughput", downloadThroughput);
+    params.put("uploadThroughput", uploadThroughput);
+
+    connectionType.ifPresent(ConnectionType -> params.put("connectionType", connectionType.get().getType()));
+
+    return new Command<>(domainName + ".emulateNetworkConditions", params);
+
+  }
+
+  /**
+   * Enables network tracking, network events will now be delivered to the client.
+   * @param maxTotalBufferSize (Optional) - Buffer size in bytes to use when preserving network payloads (XHRs, etc).
+   * @param maxResourceBufferSize (Optional) - Per-resource buffer size in bytes to use when preserving network payloads (XHRs, etc).
+   * @param maxPostDataSize (Optional) - Longest post body size (in bytes) that would be included in requestWillBeSent notification
+   * @return DevTools Command
+   */
+  public static Command<Void> enable(Optional<Integer> maxTotalBufferSize, Optional<Integer> maxResourceBufferSize, Optional<Integer> maxPostDataSize) {
+
+    Map<String, Object> params = new HashMap<>();
+
+    maxTotalBufferSize.ifPresent(integer -> params.put("maxTotalBufferSize", integer));
+    maxResourceBufferSize.ifPresent(integer -> params.put("maxResourceBufferSize", integer));
+    maxPostDataSize.ifPresent(integer -> params.put("maxPostDataSize", integer));
+
+    return new Command<>(domainName + ".enable", params);
+
+  }
+
+  /**
+   * Returns all browser cookies. Depending on the backend support, will return detailed cookie information in the cookies field
+   * @return Array of cookies
+   */
+  public static Command<Set<Cookie>> getAllCookies() {
```

Comment:
@shs96c 
In DevTools cookie the name of the fields are different (e.g expires VS expiry), this will cause the JSON deserializer to break. In addition, there are extra fields in DevTools Cookie.
What to do say about adding "asSeleniumCookie" to DevTools Cookie object. Then, the user can decide which Cookie he wants to use. Another option will be to always return the Selenium Cookie from DevTools command (and loose extra devtools cookie fields)

Reply:
It's perfectly fine to have internal classes that never leak out of the package to make serialising/deserialising easy, but public APIs should be expressed in Selenium's own classes. Put another, having the "`to/fromSeleniumCookie`" method on CDP's `Cookie` class makes sense, but it should be package-level visible, rather than public.

Reply:
added Cookies operations with Selenium Cookie.

the getCookies will be like:
`devTools.send(getAllCookies()).asSeleniumCookies()`
which will return List<org.openqa.selenium.Cookie>

the setCookie will just take Selenium Cookie.